pyi.py

```diff
@@ -246,6 +246,31 @@ def _check_platform_check(self, node: ast.Compare) -> None:
         else:
             self.error(node, Y007)
 
+    def visit_FunctionDef(self, node: ast.FunctionDef) -> None:
+        self.generic_visit(node)
+
+        for i, statement in enumerate(node.body):
+            if i == 0:
+                # normally, should just be "..."
+                if isinstance(statement, ast.Pass):
+                    # should disable this if https://github.com/python/typeshed/pull/1009 isn't
+                    # accepted
+                    self.error(statement, Y009)
+                    continue
+                elif isinstance(statement, ast.Expr) and isinstance(statement.value, ast.Ellipsis):
+                    continue
+            # allow "raise", a number of stubs have this
+            if isinstance(statement, ast.Raise):
+                continue
+            # allow assignments in constructor for now
+            # (though these should probably be changed)
+            if node.name == '__init__' and isinstance(statement, ast.Assign) and \
+                    isinstance(statement.targets[0], ast.Attribute) and \
+                    isinstance(statement.targets[0].value, ast.Name) and \
+                    statement.targets[0].value.id == 'self':
+                continue
```

Comment:
What do you think of instead disabling this case completely and not using `self.x` (or `cls.x`) assignments in typeshed at all? That would simplify things and it would mean I don't have to further complicate this condition. (Right now, there are only a few typeshed stubs that use assignments in `__init__`, and none of them use AnnAssign or `cls`.)

I don't think I can check for type comments cleanly unless we switch the AST parser to typed_ast.